### PR TITLE
Add Longhorn 1.5.1 chart with dev-2.7

### PR DESCRIPTION
**Pull request by weizhe0422**
[Longhorn 6274](https://github.com/longhorn/longhorn/issues/6274)
[Rancher 42041](https://github.com/rancher/rancher/issues/42041)

Signed-off-by: Ray Chang [ray.chang@suse.com]

cc @innobead @PhanLe1010 @khushboo-rancher @roger-ryao

### CODE REVIEW

**Comment by weizhe0422** [comment]
Try to install Longhorn `V1.5.1` via Rancher Apps but replace the image repository from rancher mirror: `rancher/mirrored-longhornio-` to `longhornio/`, the installation is successful.
image
image

Still need wait QA further verification and mirror images release: https://github.com/rancher/image-mirror/pull/418

@khushboo-rancher @roger-ryao

**Comment by roger-ryao** [comment]
verify is passed this PR as follow step

**The test steps**
1. Create pod_with_pvc. https://raw.githubusercontent.com/longhorn/longhorn/master/examples/pod_with_pvc.yaml
2. Create orphaned data.
SSH to one of the nodes and copy the contents of one replica directory, 
e.g., `pvc-abc-…012`, to a new directory named `pvc-abc-…013`. This will make `pvc-abc-…013` an orphaned replica.
3. Deploy a statefulSet and attach to the volume. https://raw.githubusercontent.com/longhorn/longhorn/master/examples/statefulset.yaml
4. Validate that the volume is created and attached to the pod properly in the Longhorn GUI.
5. Validate the pod is mounted properly.
6. Scale down and up the statefulSet to check volume detach and attach is working fine.

**Comment by innobead** [comment]
@roger-ryao have you tested the upgrade from the previous rancher chart? https://github.com/longhorn/longhorn/wiki/Release-Test-Plan---1.5.1 cc @khushboo-rancher

**Comment by roger-ryao** [comment]
> upgrade from the previous rancher chart?

Hi @innobead :
     I haven't tested the upgrade from the previous Rancher chart yet. Let me check it.

**Comment by innobead** [comment]
Should do the same as you have done at https://github.com/rancher/charts/pull/2780#issuecomment-1644984519.

**Comment by roger-ryao** [comment]
> Should do the same as you have done at [#2780 (comment)](https://github.com/rancher/charts/pull/2780#issuecomment-1644984519).

- [x] Fresh installation
- [x] Uninstallation
- [x] Upgrade from v1.4.2 - > v1.5.1
- [x] Upgrade from [v1.4.3](https://github.com/rancher/charts/pull/2780) - > v1.5.1 

**The test steps**
1. Create pod_with_pvc. https://raw.githubusercontent.com/longhorn/longhorn/master/examples/pod_with_pvc.yaml
2. Create orphaned data.
SSH to one of the nodes and copy the contents of one replica directory, 
e.g., `pvc-abc-…012`, to a new directory named `pvc-abc-…013`. This will make `pvc-abc-…013` an orphaned replica.
3. Deploy a statefulSet and attach to the volume. https://raw.githubusercontent.com/longhorn/longhorn/master/examples/statefulset.yaml
4. Validate that the volume is created and attached to the pod properly in the Longhorn GUI.
5. Validate the pod is mounted properly.
6. Scale down and up the statefulSet to check volume detach and attach is working fine.

File: charts/longhorn/102.3.0+up1.5.1/templates/validate-install-crd.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# {{- set $found "longhorn.io/v1beta2/Orphan" false -}}
+# {{- set $found "longhorn.io/v1beta1/RecurringJob" false -}}
+# {{- set $found "longhorn.io/v1beta1/Replica" false -}}
+# {{- set $found "longhorn.io/v1beta1/Setting" false -}}
+# {{- set $found "longhorn.io/v1beta1/ShareManager" false -}}
+# {{- set $found "longhorn.io/v1beta2/Snapshot" false -}}
+# {{- set $found "longhorn.io/v1beta2/SupportBundle" false -}}
+# {{- set $found "longhorn.io/v1beta2/SystemBackup" false -}}
+# {{- set $found "longhorn.io/v1beta2/SystemRestore" false -}}
+# {{- set $found "longhorn.io/v1beta1/Volume" false -}}
```
Comment (PhanLe1010) [inline charts/longhorn/102.3.0+up1.5.1/templates/validate-install-crd.yaml:22]: Miss the new CRD `volumeattachments.longhorn.io` https://github.com/longhorn/longhorn/blob/19e8fefd3ace7fb66c3f3521fc471b60a829b155/chart/templates/crds.yaml#L3555

File: charts/longhorn/102.3.0+up1.5.1/templates/userroles.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    rbac.authorization.k8s.io/aggregate-to-admin: "true"
+rules:
+- apiGroups: [ "longhorn.io" ]
+  resources: ["volumes", "volumes/status", "engines", "engines/status", "replicas", "replicas/status", "settings",
+              "engineimages", "engineimages/status", "nodes", "nodes/status", "instancemanagers", "instancemanagers/status",
+              "sharemanagers", "sharemanagers/status", "backingimages", "backingimages/status",
+              "backingimagemanagers", "backingimagemanagers/status", "backingimagedatasources", "backingimagedatasources/status",
+              "backuptargets", "backuptargets/status", "backupvolumes", "backupvolumes/status", "backups", "backups/status",
+              "recurringjobs", "recurringjobs/status", "orphans", "orphans/status", "snapshots", "snapshots/status",
+              "supportbundles", "supportbundles/status", "systembackups", "systembackups/status", "systemrestores", "systemrestores/status"]
```
Comment (PhanLe1010) [inline charts/longhorn/102.3.0+up1.5.1/templates/userroles.yaml]: Missing resources "volumeattachments", "volumeattachments/status" https://github.com/longhorn/longhorn/blob/19e8fefd3ace7fb66c3f3521fc471b60a829b155/chart/templates/clusterrole.yaml#L45C15-L45C62

File: charts/longhorn/102.3.0+up1.5.1/templates/userroles.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    rbac.authorization.k8s.io/aggregate-to-edit: "true"
+rules:
+- apiGroups: [ "longhorn.io" ]
+  resources: ["volumes", "volumes/status", "engines", "engines/status", "replicas", "replicas/status", "settings",
+              "engineimages", "engineimages/status", "nodes", "nodes/status", "instancemanagers", "instancemanagers/status",
+              "sharemanagers", "sharemanagers/status", "backingimages", "backingimages/status",
+              "backingimagemanagers", "backingimagemanagers/status", "backingimagedatasources", "backingimagedatasources/status",
+              "backuptargets", "backuptargets/status", "backupvolumes", "backupvolumes/status", "backups", "backups/status",
+              "recurringjobs", "recurringjobs/status", "orphans", "orphans/status", "snapshots", "snapshots/status",
+              "supportbundles", "supportbundles/status", "systembackups", "systembackups/status", "systemrestores", "systemrestores/status"]
```
Comment (PhanLe1010) [inline charts/longhorn/102.3.0+up1.5.1/templates/userroles.yaml]: Missing resources "volumeattachments", "volumeattachments/status" https://github.com/longhorn/longhorn/blob/19e8fefd3ace7fb66c3f3521fc471b60a829b155/chart/templates/clusterrole.yaml#L45C15-L45C62

File: charts/longhorn/102.3.0+up1.5.1/templates/userroles.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    rbac.authorization.k8s.io/aggregate-to-view: "true"
+rules:
+- apiGroups: [ "longhorn.io" ]
+  resources: ["volumes", "volumes/status", "engines", "engines/status", "replicas", "replicas/status", "settings",
+              "engineimages", "engineimages/status", "nodes", "nodes/status", "instancemanagers", "instancemanagers/status",
+              "sharemanagers", "sharemanagers/status", "backingimages", "backingimages/status",
+              "backingimagemanagers", "backingimagemanagers/status", "backingimagedatasources", "backingimagedatasources/status",
+              "backuptargets", "backuptargets/status", "backupvolumes", "backupvolumes/status", "backups", "backups/status",
+              "recurringjobs", "recurringjobs/status", "orphans", "orphans/status", "snapshots", "snapshots/status",
+              "supportbundles", "supportbundles/status", "systembackups", "systembackups/status", "systemrestores", "systemrestores/status"]
```
Comment (PhanLe1010) [inline charts/longhorn/102.3.0+up1.5.1/templates/userroles.yaml]: Missing resources "volumeattachments", "volumeattachments/status" https://github.com/longhorn/longhorn/blob/19e8fefd3ace7fb66c3f3521fc471b60a829b155/chart/templates/clusterrole.yaml#L45C15-L45C62

**Review by PhanLe1010 — CHANGES_REQUESTED** [review]

**Comment by weizhe0422** [comment]
@PhanLe1010 I've update the PR to include the new CRD: `VolumeAttachment`. 

I also refer to [Roger's test steps](https://github.com/rancher/charts/pull/2797#issuecomment-1645310891) to do the upgrade testing from `1.4.2` to `1.5.1`, and doing some tests:
 1. Install `LH 1.4.2` in `Rancher 2.7`
 2. Deploy Pod and related PVC: https://raw.githubusercontent.com/longhorn/longhorn/master/examples/pod_with_pvc.yaml
 3. Writing some data into the mount point
 4. Upgrade to `LH 1.5.1` in Rancher
 5. Deploy StatefulSet: https://raw.githubusercontent.com/longhorn/longhorn/master/examples/statefulset.yaml
 6. Make sure all of the workloads works well
    - volumes could attach to the node
    - data is intact after upgrading
    - new workload could be deployed normally
    - Scale down and up the statefulSet to check volume detach and attach is working fine
  
image

**Review by PhanLe1010 — APPROVED** [review]
LGTM

**Review by rohitsakala — DISMISSED** [review]

File: index.yaml [hunk 5]
```diff
@@ -3233,6 +3277,21 @@ entries:
...
+    apiVersion: v1
+    appVersion: v1.4.3
+    created: "2023-07-17T12:05:59.942611+08:00"
+    description: Installs the CRDs for longhorn.
+    digest: 00986f4afd2def8213d22c03e7662eb08999ea8769be599de6772e79c8865f4e
+    name: longhorn-crd
+    type: application
+    urls:
+    - assets/longhorn-crd/longhorn-crd-102.2.2+up1.4.3.tgz
+    version: 102.2.2+up1.4.3
```
Comment (rohitsuse) [inline index.yaml]: This is a PR for 102.3.0+up1.5.1 but you are adding for 102.2.2+up1.4.3. 

Can you maybe rebase this PR ?

**Review by rohitsakala — DISMISSED** [review]

File: index.yaml (hunk 5 above)
Comment (rohitsuse) [reply index.yaml]: Outdated Comment.

**Review by rohitsakala — APPROVED** [review]

**Pull request merged**